hack/ci/fetch_kubebuilder_release_bin.sh

```diff
@@ -0,0 +1,126 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# This shell fetches kubebuilder release binary and puts it under
+# /tmp/kubebuilder_bin/kubebuilder/bin/
+# Then sets KUBEBUILDER_ASSETS to that path to make go test using
+# envtest package in kubernetes-sigs/controller-runtime work properly.
+#
+# This script is largely based on CI script file "chek-everything.sh"
+# of kubernetes-sigs/controller-runtime
+
+# Version of kubebuilder release binary which this script attempts to fetch
+kubebuilder_release_version="2.3.1"
+
+goos=$(go env GOHOSTOS)
+goarch=$(go env GOHOSTARCH)
+
+# Supported arch under above release version
+# See source code of standard library "build" ("go/build") for os&arch constant
+os_darwin='darwin'
+darwin_supported_arch=('amd64')
+
+# Supported arch under above release version
+# See source code of standard library "build" ("go/build") for os&arch constant
+os_linux='linux'
+linux_supported_arch=('amd64' 'arm64' 'ppc64le')
+
+kubebuilder_release_base_url='https://github.com/kubernetes-sigs/kubebuilder/releases/download'
+fetch_url_base="${kubebuilder_release_base_url}/v${kubebuilder_release_version}"
+
+
+# Fetch kubebuilder release binaries and place them under "<argument>/kubebuilder/bin/"
+function fetch_kb_bin () {
+  local dest_dir="${1}/kubebuilder"
+  local archive_file
+  local checksum_file="checksums.txt"
+  local found
+  
+  # Check whether kubebuilder release binary is provided for current environment's OS & Arch
+  if [[ ${goos} == ${os_darwin} ]]; then
+    for (( i=0; i < ${#darwin_supported_arch[@]}; i++ )); do
+      if [[ ${goarch} == ${darwin_supported_arch[${i}]} ]]; then
+        found="true"
+        break
+      fi
+    done
+  
+    if [[ -z ${found} ]]; then
+      echo "No kubebuilder supported arch under ${goos}: ${goarch}"
+      exit 1
+    fi
+  
+  elif [[ ${goos} == ${os_linux} ]]; then
+    for (( i=0; i < ${#linux_supported_arch[@]}; i++ )); do
+      if [[ ${goarch} == ${linux_supported_arch[${i}]} ]]; then
+        found="true"
+        break
+      fi
+    done
+  
+    if [[ -z ${found} ]]; then
+      echo "No kubebuilder supported arch under ${goos}: ${goarch}"
+      exit 1
+    fi
+  
+  else
+    echo "Not kubebuilder supported os: ${goos}"
+    exit 1
+  fi
+  
+  archive_file="kubebuilder_${kubebuilder_release_version}_${goos}_${goarch}.tar.gz"
+  
+  mkdir -p ${dest_dir}
+
+  # Fetch kubebuilder release binary
+  curl -fsL "${fetch_url_base}/${archive_file}" -o "/tmp/${archive_file}"
+  if [[ ${?} != 0 ]]; then
+    echo "Failed to curl kubebuilder release binary"
```

Comment:
Nit: I do wonder whether it would be easier to just log "is kubebuilder supported on ${goos}_${goarch}" here and skip the per-platform tests !